X16 Reference - Appendix F - 65C816 Processor.md

```diff
@@ -0,0 +1,2517 @@
+
+```text
+SYNTAX           MODE       HEX LEN CYCLES      FLAGS   
+TAX              imp        AA  1   2           n.....z. .
+```
+
+Copies the contents of .A to .X.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TAY
+
+**Transfer Accumulator to Index Y**
+
+```text
+SYNTAX           MODE       HEX LEN CYCLES      FLAGS   
+TAY              imp        A8  1   2           n.....z. .
+```
+
+Copies the contents of .A to .Y.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TCD
+
+**Transfer C Accumulator to Direct Register**
+
+```text
+SYNTAX           MODE       HEX LEN CYCLES      FLAGS   
+TCD              imp        5B  1   2           n.....z. .
+```
+
+This copies the 16-bit value from the 16-bit Accumulator to the Direct Register,
+allowing you to relocate Direct Page anywhere in the first 64K of RAM.
+
+This is one of the times that the 16-bit Accumulator is called .C, as it always
+operates on a 16-bit value, regardless of the state of the **m** flag.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TCS
+
+**Transfer C Accumulator to Stack Pointer**
+
+```text
+SYNTAX           MODE       HEX LEN CYCLES      FLAGS   
+TCS              imp        1B  1   2           ........ .
+```
+
+This copies the 16-bit value from the 16-bit Accumulator to the Stack Pointer,
+allowing you to relocate the stack anywhere in the first 64K of RAM.
+
+This is one of the times that the 16-bit Accumulator is called .C, as it always
+operates on a 16-bit value, regardless of the state of the **m** flag.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TDC
+
+**Transfer Direct Register to C Accumulator**
+
+```text
+SYNTAX           MODE       HEX LEN CYCLES      FLAGS   
+TDC              imp        7B  1   2           n.....z. .
+```
+
+Copies the value of the Direct Register to the Accumulator.
+
+This is one of the times that the 16-bit Accumulator is called .C, as it always
+operates on a 16-bit value, regardless of the state of the **m** flag.
+
+
+[[Opcodes](#instructions-by-opcode)] [[By Name](#instructions-by-name)] [[By Category](#instructions-by-category)]
+
+---
+
+### TRB
+
+**Test and Reset Bit**
+
+```text
+SYNTAX           MODE       HEX LEN CYCLES      FLAGS   
+TRB $20          dir        14  2   7-2*m+w     ......z. .
+TRB $1234        abs        1C  3   8-2*m       ......z. .
+```
+
+TRB clears bits in memory, based on the bits that were set in the accumulator.
+This can be used to selectively clear or reset specific bits in a memory value.
+
+TRB also performs a bitwise AND with a memory value and the Accumulator. When
+the result of the AND operation is Zero, **z** is set.
+
```

Comment:
The order of this explanation may be misleading.  The bitwise AND occurs before the clear operation.

Reply:
ok, expanded explanation and added example.